src/main/java/org/assertj/core/api/Array2DAssert.java

```diff
@@ -0,0 +1,116 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License is distributed on
+ * an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations under the License.
+ *
+ * Copyright 2012-2020 the original author or authors.
+ */
+package org.assertj.core.api;
+
+/**
+ * Assertions applicable to two-dimensional arrays,
+ *
+ * @param <SELF> the "self" type of this assertion class. Please read &quot;<a href="http://bit.ly/1IZIRcY"
+ *          target="_blank">Emulating
+ *          'self types' using Java Generics to simplify fluent API implementation</a>&quot; for more details.
+ * @param <ELEMENT> the type of elements of the "actual" value.
+ *
+ * @author Yvonne Wang
+ * @author Alex Ruiz
+ * @author Mikhail Mazursky
+ * @author Nicolas François
+ * @author Maciej Wajcht
+ */
+public interface Array2DAssert<SELF extends Array2DAssert<SELF, ELEMENT>, ELEMENT> {
+
+  /**
+   * Verifies that the actual group of values is {@code null} or empty.
+   * <p>
+   * Example:
+   * <pre><code class='java'> // assertions will pass
+   * assertThat(new int[][] { }).isNullOrEmpty();
+   *
+   * // assertions will fail
+   * assertThat(new String[][] { {&quot;a&quot;}, {&quot;b&quot;} }).isNullOrEmpty();</code></pre>
+   *
+   * @throws AssertionError if the actual group of values is not {@code null} or not empty.
+   */
+  void isNullOrEmpty();
+
+  /**
+   * Verifies that the actual group of values is empty.
+   * <p>
+   * Example:
+   * <pre><code class='java'> // assertions will pass
+   * assertThat(new int[][] { {} }).isEmpty();
+   *
+   * // assertions will fail
+   * assertThat(new String[][] { {&quot;a&quot;}, {&quot;b&quot;} }).isEmpty();</code></pre>
+   *
+   * @throws AssertionError if the actual group of values is not empty.
+   */
+  void isEmpty();
+
+  /**
+   * Verifies that the actual group of values is not empty.
+   * <p>
+   * Example:
+   * <pre><code class='java'> // assertions will pass
+   * assertThat(new String[][] { {&quot;a&quot;}, {&quot;b&quot;} }).isNotEmpty();
+   *
+   * // assertions will fail
+   * assertThat(new int[][] { }).isNotEmpty();</code></pre>
+   *
+   * @return {@code this} assertion object.
+   * @throws AssertionError if the actual group of values is empty.
+   */
+  SELF isNotEmpty();
+
+  /**
+   * Verifies that the number of values in the actual group is equal to the given one.
```

Comment:
replace group by array.
it is also not crystal clear that the all the second dimension arrays size must be expectedSecondDimensionSize, let's state it clearly and show a failing assertions like
```
assertThat(new int[][] { {1, 2, 3}, {4, 5, 6, 7} }).hasSize(2, 3);
```

Reply:
done